lib/puppet/provider/package/pip.rb

```diff
@@ -226,6 +292,28 @@ def install_options
     join_options(@resource[:install_options])
   end
 
+  def insync?(is)
+    return false unless is && is != :absent
+    return false unless String === is && String === @resource[:ensure]
```

Comment:
Do we need this check? seems that the `:absent` case is already handled and `resouce[:ensure]` is always a String